### ISSUE TITLE
Student not automatically enrolled in Helpdesk Course

### ISSUE THREAD

**Issue by lassnj** [center-for-learning-management/moodle-local_edusupport#9]
I added sarah as a student to Testschule 1. She was not autoamtically enrolled in the helpdesk course.
All users of the school should be automatically enrolled in the helpdesk course same as the digitaler Schulhof. When I remove a user from the school, he/she should also automatically be set inactive in the helpdesk course.

**Comment by rschrenk** [comment]
I did the following steps:
1. Checked the participants list of the Help Desk of Testschool 9, Anna Lehrerin was there
2. Removed Anna Lehrerin from Testschool 9
3. Checked the participants list of the Help Desk of Testschool 9, Anna Lehrerin was not there
4. Added Anna Lehrerin to Testschool 9 as student
5. Checked the participants list of the Help Desk of Testschool 9, Anna Lehrerin was there with student permissions
6. Changed role of Anna Lehrerin in Testschool 9 to manager
7. Checked the participants list of the Help Desk of Testschool 9, Anna Lehrerin was there with student permissions

Bug confirmed:
We would intend that as a manager she has editingteacher permissions in this course.

**Comment by rschrenk** [comment]
The role was only set correctly, when the user was enrolled to the course the first time, but not when the role only changed.
fixed in 209378ed0e581064f4dc43ffde50e82527cce880

**Comment by lassnj** [comment]
When removing a user from the school, the forum posts remain with the user correctly assigned. The manager can click the view profile link, but does not have permission to view the profile anymore. 
Add exception handling for this error.
```
line 498 of /lib/setuplib.php: moodle_exception thrown
line 130 of /user/view.php: call to print_error()
```
